### PR TITLE
Update History.md with changes since last release

### PR DESCRIPTION
**Pull request by macournoyer**
Left out internal changes that don't have user impact.

There was one security fix (#1357).

### CODE REVIEW

**Review by dylanahsmith — COMMENTED** [review]
Thanks for helping prepare for a new liquid release.   It has been too long, so there have been multiple requests for a new release already (#1299, #1243, #1324, #1012)

File: History.md [hunk 1]
```diff
@@ -2,9 +2,36 @@
...
+* Fix how Template overrides static registers when #render is invoked (#1258) [Feken Baboyan]
+* Handle disabled tags errors like other liquid errors (#1275) [Dylan Thacker-Smith]
+* Fix render length resource limit so it doesn't multiply nested output (#1285) [Dylan Thacker-Smith]
+* Fix duplication of text in raw tags (#1304) [Peter Zhu]
+* Fix strict parsing of find variable with a name expression (#1317) [Dylan Thacker-Smith]
+* Use monotonic time to measure durations in Liquid::Profiler (#1362) [Dylan Thacker-Smith]
+* Fix template name in profile result for render tag timing objects (#1363) [Dylan Thacker-Smith]
+* Fix catastrophic backtracking in `RANGES_REGEX` regular expression (#1357) [Dylan Thacker-Smith]
+
+### Changed
```
[the remark this answers is not in the document]
Comment (macournoyer) [reply History.md:21]: Ah yeah!

[the remark this answers is not in the document]
Comment (dylanahsmith) [reply History.md:21]: I think the last one could be "Changes".  Possibly we could call them "Breaking Changes" and be clearer about what code they break, but I think they have a low probability of breaking code for most users of the library.

[the remark this answers is not in the document]
Comment (macournoyer) [reply History.md]: Ended up using:

- Features
- Fixes
- Breaking Changes
- Performance Improvements

All pluralized so it's more harmonious.

**Review by dylanahsmith — APPROVED** [review]

**Comment by ashmaroli** [comment]
@macournoyer Thank you for curating all the changes. However, you have missed the fact that support for Ruby 2.4 has been dropped. The next version requires at least Ruby 2.5

Additionally, not all internal changes have zero user-impact.
For example the change from `Liquid::Expression::MethodLiteral` to `Liquid::Condition::MethodLiteral` breaks the implicit contract of a public constant. Even if `Liquid::Expression::MethodLiteral` was not meant to be used outside Liquid, not explicitly using `private_constant` means that it can be considered as a public constant.

**Comment by macournoyer** [comment]
> you have missed the fact that support for Ruby 2.4 has been dropped. The next version requires at least Ruby 2.5

It's here: https://github.com/Shopify/liquid/pull/1374/files#diff-abfa5988643af1b8b2600aac13b273922dbb3372e021bf1d7caadfe7473c9561R24

> Additionally, not all internal changes have zero user-impact.
> For example the change from `Liquid::Expression::MethodLiteral` to `Liquid::Condition::MethodLiteral` breaks the implicit contract of a public constant.

Possible I made a bad decision. But at the same time, if we include all changes, the changelog will just end up being the commit log.

**Comment by ashmaroli** [comment]
> It's here: https://github.com/Shopify/liquid/pull/1374/files#diff-abfa5988643af1b8b2600aac13b273922dbb3372e021bf1d7caadfe7473c9561R24

:slightly_smiling_face: Is it?
That line says Ruby 2.4 or greater is supported. But **in a [later pull request](https://github.com/Shopify/liquid/commit/e495f75cc2e21791e4b37389f34f6845517a9fae), the support was changed to Ruby 2.5 or greater**


> But at the same time, if we include all changes..

That's true.
But you could cover all of those with a line saying `..and lots of internal changes.` at the end of the `Breaking Changes` section.

**Comment by macournoyer** [comment]
Aaah! Thanks for spotting these @ashmaroli 👍 Fixed in #1377

**Pull request merged**